src/SqlAst/QueryScope.php

```diff
@@ -106,10 +107,15 @@ public function getType($expression): Type
                 }
             }
 
-            foreach ($this->joinedTables as $joinedTable) {
+            foreach ($this->joinedTables as $join) {
+                $joinedTable = $join->getTable();
+
                 foreach ($joinedTable->getColumns() as $column) {
                     if ($column->getName() === $expression->getName()) {
-                        return $column->getType();
+                        if ($join->getType() === Join::TYPE_OUTER) {
+                            return TypeCombinator::addNull($column->getType());
+                        }
+                        return TypeCombinator::removeNull($column->getType());
```

Comment:
I think we should only add/remove NULLs for columns which are part of the join condition, but not regular columns selected from joined tables

Reply:
before doing so, we may wait a bit for sqlftw stabilization as we would need a deep dive into the AST

<img width="439" alt="grafik" src="https://user-images.githubusercontent.com/120441/221037663-717096de-ec7e-4aee-a26d-9c76c0a564c4.png">